### PR TITLE
Feature/leasing history

### PR DESCRIPTION
**Pull request by tomshudev**

### CODE REVIEW

File: src/app/core/models/leasing.model.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export type Leasing = {
+  leasingID: string;
+  transactionID: string;
+  lessor: User;
+  lessee: User;
+  product: Product;
+  status: LeasingStatusFromServer;
+  startDate: Date;
+  endDate: Date;
+  price: number;
```
Comment (Sagie501) [inline src/app/core/models/leasing.model.ts:14]: Redundant. The price is located under the product

File: src/app/core/models/leasing.model.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { User } from './user.model';
+import { Product } from './product.model';
+import { LeasingStatusFromServer } from 'src/app/shared/components/order-status/order-status.component';
+
+export type Leasing = {
+  leasingID: string;
+  transactionID: string;
+  lessor: User;
```
Comment (Sagie501) [inline src/app/core/models/leasing.model.ts:8]: Redundant. The lessor is the owner of the product.

File: src/app/core/services/products/products.service.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductsService {
+  products: Product[];
+
+  constructor() {}
+
+  getProducts() {}
```
Comment (Sagie501) [inline src/app/core/services/products/products.service.ts:12]: Shouldn't be as selector from the store?

File: src/app/core/services/leasing/leasing.service.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@Injectable({
+  providedIn: 'root',
+})
+export class LeasingService {
+  leasings: Leasing[];
+  pendingLeasings: Leasing[];
+
+  constructor() {}
+
+  getLeasings() {
```
Comment (Sagie501) [inline src/app/core/services/leasing/leasing.service.ts:17]: Shouldn't be as selector from the store?

File: src/app/shared/components/order-status/order-status.component.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  NEED_TO_RETURN = 'NEED_TO_RETURN',
+  RETURNED = 'RETURNED',
+}
+
+export enum LeasingStatusClient {
+  WAITING_FOR_APPROVE = 'Waiting for Approve',
+  DENIED = 'Denied',
+  CANCELED = 'Cancelled',
+  WAITING_FOR_DELIVERY = 'Waiting for Delivery',
+  IN_DELIVERY = 'In delivery',
```
Comment (Sagie501) [inline src/app/shared/components/order-status/order-status.component.ts:19]: In **D**elivery

File: src/app/shared/components/order-status/order-status.component.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+export enum LeasingStatusFromServer {
```
Comment (Sagie501) [inline src/app/shared/components/order-status/order-status.component.ts:3]: Maybe all this enums and functions need to be in other file?

File: src/app/shared/components/user-name-circle/user-name-circle.component.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  hashCode(str) {
+    // java String#hashCode
+    var hash = 0;
+    for (var i = 0; i < str.length; i++) {
+      hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    }
+    return hash;
+  }
+
+  getUserColor(name: string) {
```
Comment (Sagie501) [inline src/app/shared/components/user-name-circle/user-name-circle.component.ts:36]: What the FUCK 😂

File: src/app/core/models/leasing.model.ts (hunk 1 above)
Comment (tomshudev) [reply src/app/core/models/leasing.model.ts:14]: Done

**Pull request merged**